pkg/webui/account/containers/profile-settings-form/index.js

```diff
@@ -0,0 +1,292 @@
+  // Discard invalid values
+  if (matches.length < 3) return null
+
+  // Return value with correct schema for the API request
+  return { embedded: { data: matches[2], mime_type: matches[1] } }
+}
+
+const imageDataDecoder = value => {
+  if (!Boolean(value) || isGravatarProfilePicture(value)) {
+    return null
+  }
+
+  return Boolean(value.embedded)
+    ? imageUriFromEmbedded(value)
+    : getClosestProfilePictureBySize(value, 128)
+}
+
+const imageUriFromEmbedded = value =>
+  `data:${value.embedded.mime_type};base64,${value.embedded.data}`
+
+const ProfileEditForm = () => {
+  const dispatch = useDispatch()
+
+  const user = useSelector(selectUser)
+  const useGravatarConfig = useSelector(selectUseGravatarConfiguration)
+  const disableUploadConfig = useSelector(selectDisableUploadConfiguration)
+  const initialProfilePictureSource =
+    useGravatarConfig && isGravatarProfilePicture(user.profile_picture) ? 'gravatar' : 'upload'
+
+  const [profilePictureSource, setProfilePictureSource] = useState(initialProfilePictureSource)
+  const [error, setError] = useState()
+  const [emailAddress, setEmailAddress] = useState(user.primary_email_address)
+
+  const { debouncedFunction: debouncedSetEmail, cancel: debounceCancel } = debounce(
+    setEmailAddress,
+    500,
+  )
+
+  const validationContext = {
+    disableUploadConfig,
+    useGravatarConfig,
+    initialProfilePictureSource,
+  }
+
+  useEffect(() => () => debounceCancel(), [debounceCancel])
+
+  const handleEmailAddressChange = useCallback(
+    value => {
+      debouncedSetEmail(value)
+    },
+    [debouncedSetEmail],
+  )
+
+  const gravatarPreview = useMemo(
+    () =>
+      convertUriToProfilePicture(
+        `https://gravatar.com/avatar/${md5(emailAddress.toLowerCase())}?s=256&d=identicon`,
+      ),
+    [emailAddress],
+  )
+
+  const handleSubmit = useCallback(
+    async (values, { setSubmitting, resetForm }) => {
+      setError(undefined)
+      let patch = diff(user, validationSchema.cast(values, { context: validationContext }), [
+        '_profile_picture_source',
+      ])
+      if (Object.keys(patch).length === 0) {
+        patch = { ...values }
+        delete patch.profile_picture
+        delete patch._profile_picture_source
+      }
+      try {
+        const updatedUser = await dispatch(
+          attachPromise(updateUser({ id: user.ids.user_id, patch })),
+        )
+        resetForm({
+          values: {
+            ...user,
+            ...updatedUser,
+            _profile_picture_source: values._profile_picture_source,
+          },
+        })
+        toast({
+          title: sharedMessages.success,
+          message: m.successMessage,
+          type: toast.types.SUCCESS,
+        })
+      } catch (error) {
+        setError(error)
+        setSubmitting(false)
+      }
+    },
+    [dispatch, user, validationContext],
+  )
+
+  const handleDelete = useCallback(async () => {
+    try {
+      await dispatch(attachPromise(deleteUser)())
```

Comment:
This looks weird

Reply:
Please be more constructive. "This looks weird" is not actionable.
```suggestion
      await dispatch(attachPromise(deleteUser()))
```
Would be an alternative here, it doesn't really matter.

Reply:
Here is more constructive suggestion:

Please consider this gist:
```js
// outside of the component
const promisifiedDeleteUser = attachPromise(deleteUser)

...

// in delete callback
await dispatch(promisifiedDeleteUser())
```